src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/util/ASTUtils.java

```diff
@@ -197,9 +200,38 @@ public static String makeId(String fullPath) {
         return MAKE_ID.matcher(fullPath).replaceAll("-");
     }
 
-    public static String makeId(AnnotationExpr annotation) {
-        int line = annotation.getBegin().orElseThrow(RuntimeException::new).line;
-        return makeId(getNodeFullyQualifiedName(annotation.getParentNode()) + "." + annotation.getNameAsString() + "-L" + line);
+    public static String makeId(AnnotationExpr annotation, NodeWithAnnotations<?> nodeWithAnnotations) {
+        String annotationContext = getAnnotationContext(nodeWithAnnotations);
+
+        String idSuffix = "-" + annotationContext;
```

Comment:
We probably don't want to append 'null', maybe move this append into an 'else' clause?

Reply:
If `annotationContext` is `null` or empty, `idSuffix` gets overwritten with `-L<int>` in the `if` block below. So, `-null` should never be used but if it improves readability, I'll move it to `else` clause.

Reply:
Great! Reviewing on my phone I missed that!